redis/redisLimiter.go

```diff
@@ -98,34 +98,40 @@ func (rl *rateLimiter) CheckAllowedAndDecreaseTrials(key string) (*RateLimiterRe
 }
 
 func (rl *rateLimiter) rateLimit(ctx context.Context, key string) (*RateLimiterResult, error) {
-	initRemaining := int64(rl.maxFailures - 1)
-	allowed := true
-
 	rl.mutStorer.Lock()
 	defer rl.mutStorer.Unlock()
-	firstTry, err := rl.storer.SetEntryIfNotExisting(ctx, key, initRemaining, rl.limitPeriod)
+
+	totalRetries, err := rl.storer.Increment(ctx, key)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = rl.storer.SetExpireIfNotExisting(ctx, key, rl.limitPeriod)
 	if err != nil {
 		return nil, err
 	}
 
+	firstTry := totalRetries == 1
```

Comment:
i'm thinking if we might be able to avoid using Delete in `Reset` and let the key expire based on it's TTL:
- check first try before setting expire time; if it's first try, set Expire to limitPeriod anyway
- if it's not first try, set Expire only if not existing
- on Reset set key to 0 and do not change expire time

Reply:
applied